target/snitch_cluster/sw/toolchain.mk

```diff
@@ -42,6 +42,12 @@ RISCV_CFLAGS += -O3
 ifeq ($(DEBUG), ON)
 RISCV_CFLAGS += -g
 endif
+
+ifeq ($(SELECT_PLATFORM), gvsoc)
+RISCV_CFLAGS += -D__GVSOC__
+endif
+
+
```

Comment:
This should also not be necessary anymore. Rather when compiling we have to make sure to reference the sources in the GVSoC-specific runtime. See:

https://github.com/pulp-platform/snitch_cluster/blob/13d1e6b3aaa8e53d5b39858c6ee871b186bf9aa2/target/snitch_cluster/sw.mk#L8-L14

Reply:
Actually this is not specific to gvsoc, it's something generic which can be used on any platform, but in particular on boards with openocd. Gvsoc can support any kind of printf, currently both the tohost/fromhost support of the runtime and also openocd.
Probably the name is misleading, I can rename it to SELECT_IO which would by default be htif, and could be overwritten to ocd ?

Reply:
I see, thanks for the clarification. In that case I agree, and I would do two things, if you think it would make sense:
- move the OpenOCD related functions (e.g. `__semihost`) to an `openocd.h` file which would be included by a Snitch runtime implementation only if semihosting is supported on that target
- rename the `__GVSOC__` flag to something like `OPENOCD_SEMIHOSTING`

Reply:
Yes it makes sense, I've pushed the modifications, let me know.